src/main/java/com/onarandombox/MultiverseCore/utils/PermissionsTool.java

```diff
@@ -0,0 +1,329 @@
+import com.onarandombox.MultiverseCore.destination.ParsedDestination;
+import org.bukkit.command.CommandSender;
+import org.bukkit.permissions.Permission;
+import org.bukkit.permissions.PermissionDefault;
+import org.bukkit.plugin.PluginManager;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * A utility class for registering and checking permissions.
+ */
+public class PermissionsTool {
+    private static final String TELEPORT_PERM_PREFIX = "multiverse.teleport.";
+
+    private final MultiverseCore plugin;
+    private PluginManager pluginManager;
+    private Permission worldAccessPermission;
+    private Permission bypassGameModePermission;
+    private Permission bypassEntryFeePermission;
+    private Permission bypassPlayerLimitPermission;
+
+    public PermissionsTool(MultiverseCore plugin) {
+        this.plugin = plugin;
+    }
+
+    /**
+     * Sets up the permissions for Multiverse.
+     */
+    public void setUpPermissions() {
+        this.pluginManager = this.plugin.getServer().getPluginManager();
+        setUpWorldPermissionWildcards();
+    }
+
+    /**
+     * Set up the world permissions wildcards.
+     */
+    private void setUpWorldPermissionWildcards() {
+        this.worldAccessPermission = new Permission("multiverse.access.*",
+                "Allows access to all worlds",
+                PermissionDefault.OP);
+        this.bypassGameModePermission = new Permission("mv.bypass.gamemode.*",
+                "Allows players with this permission to ignore gamemode changes.",
+                PermissionDefault.FALSE);
+        this.bypassEntryFeePermission = new Permission("multiverse.exempt.*",
+                "A player who has this does not pay to enter this world, or use any MV portals in it",
+                PermissionDefault.OP);
+        this.bypassPlayerLimitPermission = new Permission("mv.bypass.playerlimit.*",
+                "A player who can enter this world regardless of whether its full",
+                PermissionDefault.OP);
+
+        try {
+            pluginManager.addPermission(this.worldAccessPermission);
+            pluginManager.addPermission(this.bypassGameModePermission);
+            pluginManager.addPermission(this.bypassEntryFeePermission);
+            pluginManager.addPermission(this.bypassPlayerLimitPermission);
+        } catch (IllegalArgumentException e) {
+            Logging.finer("World permissions wildcard already registered.");
+        }
+
+        Logging.finer("Registered world permissions wildcard.");
+    }
+
+    /**
+     * Registers the permissions for a world.
+     *
+     * @param world The world to register permissions for.
+     */
+    public void registerMVWorldPermissions(MVWorld world) {
+        Permission accessPermission = new Permission("multiverse.access." + world.getName(),
+                "Allows access to " + world.getName(),
+                PermissionDefault.OP);
```

Comment:
Same DRYing could be done with creation of permissions, methods to reduce the repeated stuff in these lines.